packages/ui/src/components/atoms/Popover/Popover.tsx

```diff
@@ -10,21 +10,34 @@ const PopoverTrigger = PopoverPrimitive.Trigger;
 
 const PopoverContent = React.forwardRef<
   React.ElementRef<typeof PopoverPrimitive.Content>,
-  React.ComponentPropsWithoutRef<typeof PopoverPrimitive.Content>
->(({ className, align = 'center', sideOffset = 4, ...props }, ref) => (
-  <PopoverPrimitive.Portal>
+  React.ComponentPropsWithoutRef<typeof PopoverPrimitive.Content> & { disablePortal?: boolean }
+>(({ className, align = 'center', sideOffset = 4, disablePortal = false, ...props }, ref) =>
+  disablePortal ? (
     <PopoverPrimitive.Content
       ref={ref}
       align={align}
       sideOffset={sideOffset}
       className={ctw(
-        'bg-popover text-popover-foreground animate-in data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 z-50 rounded-md border p-4 shadow-md outline-none',
+        'bg-popover text-popover-foreground animate-in data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 z-[100] rounded-md border p-4 shadow-md outline-none',
         className,
       )}
       {...props}
     />
-  </PopoverPrimitive.Portal>
-));
+  ) : (
+    <PopoverPrimitive.Portal>
+      <PopoverPrimitive.Content
+        ref={ref}
+        align={align}
+        sideOffset={sideOffset}
+        className={ctw(
+          'bg-popover text-popover-foreground animate-in data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 z-[100] rounded-md border p-4 shadow-md outline-none',
+          className,
+        )}
+        {...props}
+      />
+    </PopoverPrimitive.Portal>
+  ),
+);
```

Comment:
Took me a bit to see the only difference is the portal. Also, in what case we don't want a portal?
```suggestion
(({ className, align = 'center', sideOffset = 4, disablePortal = false, children, ...props }, ref) => {
  const Popover: FunctionComponent<ComponentProps<typeof PopoverPrimitive.Content>> = ({children, ...props}) =>  (
    <PopoverPrimitive.Content
    ref={ref}
    align={align}
    sideOffset={sideOffset}
    className={ctw(
      'bg-popover text-popover-foreground animate-in data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 z-[100] rounded-md border p-4 shadow-md outline-none',
      className,
    )}
    {...props}
    >
      {children}
    </PopoverPrimitive.Content>
  );
  
  if (disablePortal) {
    return (
      <Popover {...props}>
        {children}
      </Popover>
    );
  }
  
  return (
    <PopoverPrimitive.Portal>
      <Popover {...props}>
        {children}
      </Popover>
    </PopoverPrimitive.Portal>
  );
  
  }
)
```

Reply:
P1